Sources/AppKitNavigation/AppKitAnimation.swift

```diff
@@ -0,0 +1,125 @@
+#if canImport(AppKit) && !targetEnvironment(macCatalyst)
+  import AppKit
+
+  #if canImport(SwiftUI)
+    import SwiftUI
+  #endif
+
+  import SwiftNavigation
+
+  @MainActor
+  public func withAppKitAnimation<Result>(
+    _ animation: AppKitAnimation? = .default,
+    _ body: () throws -> Result,
+    completion: (@Sendable (Bool?) -> Void)? = nil
+  ) rethrows -> Result {
+    var transaction = UITransaction()
+    transaction.appKit.animation = animation
+    if let completion {
+      transaction.appKit.addAnimationCompletion(completion)
+    }
+    return try withUITransaction(transaction, body)
+  }
+
+  public struct AppKitAnimation: Hashable, Sendable {
+    fileprivate let framework: Framework
+
+    @MainActor
+    func perform<Result>(
+      _ body: () throws -> Result,
+      completion: ((Bool?) -> Void)? = nil
+    ) rethrows -> Result {
+      switch framework {
+      case let .appKit(animation):
+        var result: Swift.Result<Result, Error>?
+        NSAnimationContext.runAnimationGroup { context in
+          context.allowsImplicitAnimation = true
+          context.duration = animation.duration
+          context.timingFunction = animation.timingFunction
```

Comment:
@Mx-Iris I've familiarized myself a bit more with this API by adding a configurable timing function, and it also occurred to me that because `withAppKitAnimation` is intended to be called from the model, where `view.animator` is not available, that it probably makes sense to always set `allowsImplicitAnimation` to `true`. Does this sound right to you?

Reply:
Yes, it's right. `NSView`/`NSWindow` implements the `NSAnimatablePropertyContainer` protocol. When the view's `frame`, `size`, or `position` changes, it can produce animated effects. Through its animator, these properties can be used transparently to generate animations. 

Setting `NSAnimationContext`'s `allowsImplicitAnimation` to true, directly modifying the view's properties will produce animations. This means something similar to UIKit's UIView.animate method. But only the `frame`, `frameSize`, and `frameOrigin` properties are supported. More property animations require a `layer-backed` view, that is, the view's `wantsLayer` = true.